### ISSUE TITLE
[Feature]: Add validation-set ranking for discovered equations

### ISSUE THREAD

**Issue by DiogoRibeiro7** [MilesCranmer/PySR#1270]
### Feature Request

PySR exposes a Pareto frontier of equations ranked by training loss, complexity, and score. For ML workflows, it would be useful to evaluate all discovered equations on a held-out validation set.

Possible API:

```python
model.score_equations(X_val, y_val, metric="neg_mean_squared_error")
```

This could return `model.equations_` with extra columns such as `validation_loss`, `validation_r2`, or `validation_score`.

Motivation:
- Select symbolic expressions by generalization, not only training loss.
- Make model selection easier for noisy datasets.
- Reduce boilerplate for evaluating every `lambda_format` expression manually.

**Comment by MilesCranmer** [comment]
> Reduce boilerplate for evaluating every lambda_format expression manually.

You can use `model.predict`, you do not need to use `lambda_format`
